src/Compilers/CSharp/Test/Semantic/Semantics/NullConditionalAssignmentTests.cs

```diff
@@ -0,0 +1,1980 @@
+
+            comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
+                // (3,12): warning CS0414: The field 'C.f' is assigned but its value is never used
+                //     string f;
+                Diagnostic(ErrorCode.WRN_UnreferencedFieldAssg, "f").WithArguments("C.f").WithLocation(3, 12));
+        }
+
+        [Fact]
+        public void LangVersion_02()
+        {
+            // The only thing we want to diagnose is a member binding expression as the LHS of any assignment.
+            // Nested assignments within conditional access args, etc. have always been allowed.
+            var source = """
+                class C
+                {
+                    public string this[string s] { get => s; set { } }
+
+                    static void M(C c)
+                    {
+                        string s = "a";
+                        _ = c?[s = "b"];
+                        c?[s = "b"] = "c"; // 1
+                    }
+                }
+                """;
+
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular13);
+            comp.VerifyEmitDiagnostics(
+                // (7,16): warning CS0219: The variable 's' is assigned but its value is never used
+                //         string s = "a";
+                Diagnostic(ErrorCode.WRN_UnreferencedVarAssg, "s").WithArguments("s").WithLocation(7, 16),
+                // (9,11): error CS8652: The feature 'null conditional assignment' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         c?[s = "b"] = "c"; // 1
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, @"[s = ""b""] = ""c""").WithArguments("null conditional assignment").WithLocation(9, 11));
+
+            comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
+                // (7,16): warning CS0219: The variable 's' is assigned but its value is never used
+                //         string s = "a";
+                Diagnostic(ErrorCode.WRN_UnreferencedVarAssg, "s").WithArguments("s").WithLocation(7, 16));
+        }
+
+        [Theory]
+        [InlineData(SyntaxKind.BarEqualsToken)]
+        [InlineData(SyntaxKind.AmpersandEqualsToken)]
+        [InlineData(SyntaxKind.CaretEqualsToken)]
+        [InlineData(SyntaxKind.LessThanLessThanEqualsToken)]
+        [InlineData(SyntaxKind.GreaterThanGreaterThanEqualsToken)]
+        [InlineData(SyntaxKind.GreaterThanGreaterThanGreaterThanEqualsToken)]
+        [InlineData(SyntaxKind.PlusEqualsToken)]
+        [InlineData(SyntaxKind.MinusEqualsToken)]
+        [InlineData(SyntaxKind.AsteriskEqualsToken)]
+        [InlineData(SyntaxKind.SlashEqualsToken)]
+        [InlineData(SyntaxKind.PercentEqualsToken)]
+        [InlineData(SyntaxKind.EqualsToken)]
+        [InlineData(SyntaxKind.QuestionQuestionEqualsToken)]
+        public void LangVersion_03(SyntaxKind kind)
```

Comment:
It occurs to me that if a new assignment kind is added later, it will be gated behind its own LangVersion check, which is at least as strict as this one.